ownCloudAppShared/AppLock/AppLockManager.swift

```diff
@@ -432,32 +436,37 @@ public class AppLockManager: NSObject {
 				passcodeViewController.view.setNeedsLayout()
 			}
 			updateLockCountdown()
-
-			lockTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(self.updateLockCountdown), userInfo: nil, repeats: true)
+			
+			if lockTimer == nil {
+				lockTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(self.updateLockCountdown), userInfo: nil, repeats: true)
+			}
 		}
 	}
 
 	@objc private func updateLockCountdown() {
-		if let date = self.lockedUntilDate {
-			let interval = Int(date.timeIntervalSinceNow)
-			let seconds = interval % 60
-			let minutes = (interval / 60) % 60
-			let hours = (interval / 3600)
-
+		if let lockDelay = self.lockDelay {
+			let hours = Int(lockDelay) / 3600
+			let minutes = Int(lockDelay) / 60 % 60
+			let seconds = Int(lockDelay) % 60
+			
+			if lockDelay > 0 {
+				self.lockDelay = (lockDelay - 1)
```

Comment:
Counting down the lock delay with a timer has several issues:
- the app needs to stay in the foreground for the entire duration of the countdown
- additional instances of `AppLockManager` running in different processes (possibly even scenes) also reduce the lock delay every second, so that if the lock screen is shown in the app, in the Share Extension and the File Provider UI at the same time, it would run a 3x speed